Client/Frontend/Login Management/LoginDataSource.swift

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+/// Data source for handling LoginData objects from a Cursor
+class LoginDataSource: NSObject, UITableViewDataSource {
+    fileprivate let emptyStateView = NoLoginsView()
```

Comment:
Usually your data layer won't have views or updating views. But, this is a UITableViewDataSource which has `cellForRowAt indexpath method`. 

Suggestion: Add a comment explaining why we have a view in data source class